libs/api/ebpf_api.cpp

```diff
@@ -2242,14 +2256,249 @@ ebpf_program_bind_map(fd_t program_fd, fd_t map_fd)
     return win32_error_code_to_ebpf_result(invoke_ioctl(request));
 }
 
+typedef struct _ebpf_ring_buffer_subscription
+{
```

Comment:
Can you annotate what fields must be accessed under the lock using [concurrency annotation](https://docs.microsoft.com/en-us/cpp/code-quality/annotating-locking-behavior?view=msvc-170)